dev/com.ibm.ws.security.oauth/src/com/ibm/ws/security/oauth20/web/CoverageMapEndpointServices.java

```diff
@@ -50,14 +51,7 @@ protected void handleEndpointRequest(OAuth20Provider provider, HttpServletReques
         if (request.getMethod().equalsIgnoreCase(HTTP_METHOD_GET) || request.getMethod().equalsIgnoreCase(HTTP_METHOD_HEAD)) {
             processHeadOrGet(provider, request, response);
         } else {
-            String errorMsg = TraceNLS.getFormattedMessage(this.getClass(),
-                    MESSAGE_BUNDLE,
-                    "OAUTH_UNSUPPORTED_METHOD",
-                    new Object[] { request.getMethod(), this.getClass().getSimpleName() },
-                    "CWWKS1433E: The HTTP method {0} is not supported for the service {1}.");
-            Tr.error(tc, errorMsg);
-
-            throw new OidcServerException(errorMsg, OIDCConstants.ERROR_SERVER_ERROR, HttpServletResponse.SC_METHOD_NOT_ALLOWED);
+            throw new OidcServerException(new BrowserAndServerLogMessage(tc, "OAUTH_UNSUPPORTED_METHOD", new Object[] { request.getMethod(), this.getClass().getSimpleName() }), OIDCConstants.ERROR_SERVER_ERROR, HttpServletResponse.SC_METHOD_NOT_ALLOWED);
```

Comment:
This is one case we hadn't talked about yet. Notice in the original changes there's a call to `Tr.error(tc, errorMsg);`. Remember that that's actually logging the error message in the server logs; that's something that we want to keep. The change as it is now would no longer log that message. I'd update this to

```Java
BrowserAndServerLogMessage errorMsg = new BrowserAndServerLogMessage(tc, "OAUTH_UNSUPPORTED_METHOD", new Object[] { request.getMethod(), this.getClass().getSimpleName() });
Tr.error(tc, errorMsg.getServerErrorMessage());
throw new OidcServerException(errorMsg, OIDCConstants.ERROR_SERVER_ERROR, HttpServletResponse.SC_METHOD_NOT_ALLOWED);
```
That will ensure the error message is still logged in the server logs.

There are several cases in this class that should do the same thing.